lighthouse-core/gather/gatherers/dobetterweb/datenow.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Tests whether the page is using Date.now().
+ */
+
+/* global window, __returnResults */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+function patchDateNow() {
+  window.__stackTraceErrors = [];
+  // Override Date.now() so we know when if it's called.
+  const orig = Date.now;
+  Date.now = function() {
+    try {
+      throw Error('__called Date.now()__');
+    } catch (e) {
+      if (e.stack) {
+        const split = e.stack.split('\n');
+        const m = split[split.length - 1].match(/(https?:\/\/.*):(\d+):(\d+)/);
```

Comment:
I tried out this date.now monkeypatch in the console, but it led to `e.stack` as

```sh
"Error: __called Date.now()__
    at Error (native)
    at Function.Date.now (<anonymous>:4:13)
    at <anonymous>:1:6"
```

The match() then returns null. I think you could make the regex testing here a bit more liberal.